card-game-client/app/components/ui/admin-card.tsx

```diff
@@ -70,9 +75,12 @@ interface AdminCardProps {
 }
 
 export default function AdminCard(categoryName: AdminCardProps) {
-  const [filteredCardData, setFilteredCardData] = useState<any[]>(categories); // change state
+  const [filteredCardData, setFilteredCardData] =
+    useState<QuizCard[]>(categories);
   const [modalIsOpen, setModalIsOpen] = useState<boolean>(false);
-  const [selectedCard, setSelectedCard] = useState<any>(null); // change type
+  const [selectedCard, setSelectedCard] = useState<QuizCard | undefined>(
```

Comment:
undefined for the initial value? Is this intentional?

Reply:
Yes, selectedCard is for when modal opens and initially no cards are selected

Reply:
undefined because implemented QuizCard type